test/max_expr_cnt/max_expr_cnt_test.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import "testing"
+
+func TestMaxExprCnt(t *testing.T) {
+	_, err := Parse("", []byte("infinite parse"), MaxExpressions(5))
+	if err == nil {
+		t.Errorf("expected non nil error message for testing max expr cnt option.")
```

Comment:
`t.Errorf` already includes `t.Fail` according to [godoc](https://godoc.org/testing#T.Errorf). Please remove `t.Fail()` calls in this file.
Why do you use `t.Errorf` in the first place (why do you want to continue executing the test after an unexpected behavior was encountered)? I suggest to change the calls to `t.Fatalf`.

Reply:
Switched to `t.Fatalf`